kong/runloop/handler.lua

```diff
@@ -643,7 +643,8 @@ do
   local function build_services_init_cache(db)
     local services_init_cache = {}
 
-    for service, err in db.services:each(nil, GLOBAL_QUERY_OPTS) do
+    local page_size = db.services.pagination.max_page_size
```

Comment:
Shouldn't this be protected against `db.services.pagination` being `nil`? In other places you have done something equivalent to:

```
local page_size = db.services.pagination and db.services.pagination.max_page_size
```